src/Umbraco.Web/HtmlStringUtilities.cs

```diff
@@ -235,10 +247,82 @@ public IHtmlString Truncate(string html, int length, bool addElipsis, bool treat
                     outputms.Position = 0;
                     using (TextReader outputtr = new StreamReader(outputms))
                     {
-                        return new HtmlString(outputtr.ReadToEnd().Replace("  ", " ").Trim());
+                        string result = String.Empty;
```

Comment:
We have some best practices and follow normal Microsoft standards (which is also enforced by Resharper), any reference to `string` should use the type qualifier of `string` not the object qualifier of `String` so this should be 

`string.Empty` similarly below `String.IsNullOrEmpty` should be `string.IsNullOrEmpty` .  On that note, this should check for `IsNullOrWhitespace` instead. We also have an extension method for this so you could do:  `firstTrim.IsNullOrWhitespace()`